exercises/risk-risiko.cpp

```diff
@@ -27,3 +27,215 @@
+    for (int i = 0; i < choice.length(); i++) // formatting the input, just in case someone types something like "TaIlS"
+      choice[i] = tolower(choice[i]);
+
+    bool coin = rand() % 2; // heads = 0, tails = 1
+    cout << endl;
+    if (coin == 0)
+      cout << "It's heads!" << endl;
+    else
+      cout << "It's tails!" << endl;
+
+    if (choice == "heads" && coin == 0)
+    {
+      cout << "You're the attacker! (Red)" << endl
+           << endl;
+      return true;
+    }
+    else if (choice == "tails" && coin == 1)
+    {
+      cout << "You're the attacker! (Red)" << endl
+           << endl;
+      return true;
+    }
+    else if (choice != "heads" && choice != "tails")
+    {
+      cout << "But you wrote neither heads nor tails... try again." << endl;
+      continue;
+    }
+    else
+    {
+      cout << "You're the defender! (Blue)" << endl
+           << endl;
+    }
+    return false;
+  } while (true);
+}
+
+int main()
+{
+  srand(time(0));
+  do
+  {
+    short scoreRed = 0, scoreBlue = 0; // players scores
+    PlayerDices Red;
+    PlayerDices Blue;
+    bool priority = coin_flip();
+    cout << "Red dices:" << endl;
+    Red.tripleDiceThrow();
+    cout << endl;
+    cout << "Blue dices:" << endl;
+    Blue.tripleDiceThrow();
+    cout << endl;
+
+    cout << "  R    B" << endl; // false/0 = defender, true/1 = attacker
+    // super cool graphics ahead! wowowowow! vvv
+    if (!Red.compareN(Blue.getN()))
+    {
+      cout << "N " << Red.getN() << " vs " << Blue.getN() << "=> blue wins!" << endl;
+      scoreBlue++;
+    }
+    else
+    {
+      cout << "N " << Red.getN() << " vs " << Blue.getN() << " => red wins!" << endl;
+      scoreRed++;
+    }
+    if (!Red.compareM(Blue.getM()))
+    {
+      cout << "M " << Red.getM() << " vs " << Blue.getM() << "=> blue wins!" << endl;
+      scoreBlue++;
+    }
+    else
+    {
+      cout << "M " << Red.getM() << " vs " << Blue.getM() << " => red wins!" << endl;
+      scoreRed++;
+    }
+    if (!Red.compareO(Blue.getO()))
+    {
+      cout << "O " << Red.getO() << " vs " << Blue.getO() << "=> blue wins!" << endl;
+      scoreBlue++;
+    }
+    else
+    {
+      cout << "O " << Red.getO() << " vs " << Blue.getO() << " => red wins!" << endl;
+      scoreRed++;
+    }
+
+    if (scoreBlue > scoreRed)
+      cout << "BLUE WINS! (" << scoreBlue << " to " << scoreRed << ")" << endl;
+    else
+      cout << "RED WINS! (" << scoreRed << " to " << scoreBlue << ")" << endl;
+
+    cout << endl;
+    cout << "Play again? (y/n)" << endl; // restart
+    do
+    {
+      char choice2;
+      cin >> choice2;
+      choice2 = tolower(choice2);
+      if (choice2 == 'y')
+      {
+        cout << "Good luck!\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n"; // looks ugly, I know
+        break;
+      }
+      else if (choice2 == 'n')
+      {
+        cout << "See you soon!" << endl;
+        return 0;
+      }
+      else
+        cerr << "Invalid input. Try again!" << endl;
```

Comment:
Anche quiz forse gli else if ed else li puoi convertire in if

Reply:
L'else lo puoi letteralmente cancellare insieme alle sue parentesi graffe, non serve

Reply:
non avevo notato di averne saltato uno! sistemo

Reply:
già nel secondo commit il penultimo else era stato levato, non so perchè non ha segnato outdated qua sopra
comunque ora dovrebbe essere sistemato